src/main/java/aspects/MetricsAspect.java

```diff
@@ -0,0 +1,48 @@
+package aspects;
+
+import beans.AuthenticatedRequestBean;
+import com.timgroup.statsd.StatsDClient;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.lang.reflect.Method;
+
+/**
+ * This aspect records various metrics for every request.
+ */
+@Aspect
+public class MetricsAspect {
+
+    @Autowired
+    protected StatsDClient datadogStatsDClient;
+
+    @Around(value = "@annotation(org.springframework.web.bind.annotation.RequestMapping)")
+    public Object logRequest(ProceedingJoinPoint joinPoint) throws Throwable {
+        Object[] args = joinPoint.getArgs();
+        String domain = "<unknown>";
+        String user = "<unknown>";
+
+        String requestPath = getRequestPath(joinPoint);
+        if (args[0] instanceof AuthenticatedRequestBean) {
+            AuthenticatedRequestBean bean = (AuthenticatedRequestBean) args[0];
+            domain = bean.getDomain();
+            user = bean.getUsernameDetail();
+        }
+
+        datadogStatsDClient.increment("requests", domain, user, requestPath);
+        long startTime = System.nanoTime();
+        Object result = joinPoint.proceed();
+        datadogStatsDClient.gauge("timings", (System.nanoTime() - startTime) / 1000000, domain, user, requestPath);
```

Comment:
these tags should be formatted with the tag name as follows `domain:{domain}` etc.